client/state/data-layer/wpcom/sites/settings/index.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SITE_SETTINGS_SAVE,
+	SITE_SETTINGS_SAVE_FAILURE,
+	SITE_SETTINGS_SAVE_SUCCESS,
+	SITE_SETTINGS_REQUEST,
+	SITE_SETTINGS_REQUEST_FAILURE,
+	SITE_SETTINGS_REQUEST_SUCCESS,
+} from 'state/action-types';
+import { receiveSiteSettings, updateSiteSettings } from 'state/site-settings/actions';
+import { normalizeSettings } from 'state/site-settings/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+
+export const fetchSiteSettings = ( { dispatch }, action ) => {
+	const { siteId } = action;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/sites/${ siteId }/settings`,
+				apiVersion: '1.1',
+			},
+			action
+		)
+	);
+};
+
+export const addSiteSettings = ( { dispatch }, action, next, { name, description, settings } ) => {
+	const { siteId } = action;
+	const savedSettings = {
+		...normalizeSettings( settings ),
+		blogname: name,
+		blogdescription: description,
+	};
+	dispatch( receiveSiteSettings( siteId, savedSettings ) );
+	dispatch( {
+		type: SITE_SETTINGS_REQUEST_SUCCESS,
+		siteId,
+	} );
```

Comment:
how much work would it take to remove the need for this action in this PR?

does `SITE_SETTINGS_REQUEST_SUCCESS` have any intrinsic meaning to the app?